### ISSUE TITLE
task 2.0

### ISSUE THREAD

**Issue by JohnKhandygo** [SergeyChernonog/ProgTechJava2018#8]
**Общая формулировка**

Разработать консольную утилиту, реализующую указанную в задании функциональность, на языках Java и/или Kotlin. При проектировании использовать объектно-ориентированный подход. Разработать автоматические тесты для отдельных частей программы. Написанный код держать в отдельном проекте или модуле в репозитории на GitHub.

**Шифрация**

Шифрация (-c) или дешифрация (-d) указанного (в аргументе командной строки) файла. Выходной файл указывается как -o filename.txt, по умолчанию имя формируется из имени входного файла с добавлением расширения.
Алгоритм шифрации XOR. Ключ указывается после -c или -d в шестнадцатеричной системе, длина ключа -- любое целое количество байт.

Command Line: ciphxor [-c key] [-d key] inputname.txt [-o outputname.txt]

Кроме самой программы, следует написать автоматические тесты к ней.

**Comment by SergeyChernonog** [comment]
Второе задание в этом репозитории https://github.com/SergeyChernonog/XORcipher

**Comment by JohnKhandygo** [comment]
Добавил в задание пункт про скрипты сборки/запуска.

**Comment by SergeyChernonog** [comment]
Еще вопрос по формулировке задания 
"Выходной файл указывается как -o filename.txt, по умолчанию имя формируется из имени входного файла с добавлением расширения."
То есть все-таки это работа с текстовым файлом ?

**Comment by JohnKhandygo** [comment]
Нет, `filename.txt` --- это просто пример. Когда Вы автоматически формируете имя выходного файла, надо просто добавить к нему какое-то расширение. Типа `.xor` или `.xr`.
